### PR TITLE
Monorepo: Move `bin/build-package.js` script to `calypso-build` package

### PR DESCRIPTION
**Pull request by ockham**
#### Changes proposed in this Pull Request

* Use tabs (instead of spaces) to indent our `package.json` files (you might wanna look at this PR with [whitespace changes filtered](https://github.com/Automattic/wp-calypso/pull/31542/files?w=1)).
* Move `bin/build-package.js` script to `calypso-build` package
* Expose as that package's binary
* Invoke that binary from other packages' `package.json` files

I think it makes sense to co-locate this build script with the config files it is (or will be) based on. Arguably, it's more similar in spirit to those generic configs (which we are going to use to build packages) than to the Calypso-specific stuff at the root level. (It is also kind of a hard requirement for package isolation.)

Factored out of #31464 to make that PR smaller.

#### Testing instructions

```
npm run distclean
npm ci
npm start
```

and verify in your browser that Calypso actually works.

Then, build Jetpack blocks

```
npx lerna bootstrap --scope='@automattic/jetpack-blocks'
npx lerna run prepublishOnly --stream --scope='@automattic/jetpack-blocks'
```

and copy them to your local Jetpack Docker instance to test them there.

Finally, test package:

```
npm run test-packages
```

### CODE REVIEW

**Comment by simison** [comment]
> Use tabs (instead of spaces) to indent our package.json files (you might wanna look at this PR with whitespace changes filtered).

Spaces in `package.json` seems intentional tho: 
https://github.com/Automattic/wp-calypso/blob/2b2ca2b01466b33add52c533cb6383cef7080be2/.editorconfig#L20-L22

Also root `package.json` uses spaces. This keeps being confusing so I'd rather see everything turned to tabs at once and editorconfig rule updated.

Spaces are used prolly just for historic reasons (NPM enforced them until NPM v5) and [nowadays NPM doesn't care](https://github.com/npm/npm/issues/5694#issuecomment-307142476).

**Comment by simison** [comment]
>Use tabs (instead of spaces) to indent our package.json files (you might wanna look at this PR with whitespace changes filtered).

I went ahead and sorted all Calypso `package.json` files in one PR: https://github.com/Automattic/wp-calypso/pull/31543

**Comment by ockham** [comment]
> > Use tabs (instead of spaces) to indent our package.json files (you might wanna look at this PR with whitespace changes filtered).
> 
> Spaces in `package.json` seem intentional tho:
> [wp-calypso/.editorconfig](https://github.com/Automattic/wp-calypso/blob/2b2ca2b01466b33add52c533cb6383cef7080be2/.editorconfig#L20-L22)
> 
> Lines 20 to 22 in [2b2ca2b](/Automattic/wp-calypso/commit/2b2ca2b01466b33add52c533cb6383cef7080be2)
> 
>  [{package.json,*.yml}] 
>  indent_size = 2 
>  indent_style = space 
> Also root `package.json` uses spaces. This keeps being confusing so I'd rather see everything turned to tabs at once and editorconfig rule updated.

Hmm. I've enabled format-on-save, and prettier keeps changing those spaces to tabs (which has at times driven me nuts when editing the root `package.json`). So I'd need to add `*.json` files to `.prettierignore` -- turns out I've [done that before](https://github.com/Automattic/wp-calypso/pull/22907), but then [that file was deleted](https://github.com/Automattic/wp-calypso/pull/24906) (:wave: @dmsnell :grimacing: ) And actually it'd be kinda nice to use prettier for json files. (Does no one else use format-on-save? Doesn't the commit hook prettify JSON files for people?)

/cc `prettier` guru @sirreal 

> Spaces are used prolly just for historic reasons (NPM enforced them until NPM v5) and [nowadays NPM doesn't care](https://github.com/npm/npm/issues/5694#issuecomment-307142476).

**Comment by ockham** [comment]
@simison Upon second reading, you were advocating _for_ tabs.
I shouldn't comment on PRs before my first :coffee: :man_facepalming:

**Comment by simison** [comment]
Running `npx lerna bootstrap --concurrency=2 --scope '@automattic/jetpack-blocks'` also seems to enforce tabs for me. Let's get this merged asap and we're good: https://github.com/Automattic/wp-calypso/pull/31543 :-)

**Comment by sirreal** [comment]
Say goodbye to your package.json woes:

https://github.com/Automattic/wp-calypso/pull/31545

File: packages/calypso-build/package.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
 	"bugs": {
 		"url": "https://github.com/Automattic/wp-calypso/issues"
 	},
+	"bin": "./bin/build-package.js",
```
Comment (sirreal) [inline packages/calypso-build/package.json]: Do you know how `npx` plays with packages with multiple bins? I don't off the top of my head.

It seems like we'll have multiple bins here eventually.

As some point we might want `cb` (Calypso build, props @blowery). I don't know if we'd want a single bin like `cb package` or `cb-package`?

We don't need to change it here, but worth thinking about.

```suggestion
	"bin": { "cb-package": "./bin/build-package.js" },
```

**Review by sirreal — COMMENTED** [review]
Left a comment, I'd like to see about this independent of the whitespace changes (#31545)

File: packages/calypso-build/package.json (hunk 1 above)
Comment (ockham) [reply packages/calypso-build/package.json]: > It seems like we'll have multiple bins here eventually.

Wasn't so sure about that. I thought building a generic package build script might cut it? Was the reason why I was going with YAGNI anyway.

Anyway, `npx` can handle multiple bins, it's something like

```
npx --package @automattic/calypso-build cb-package
```

Comment (ockham) [reply packages/calypso-build/package.json]: Want me to change it to that effect already? Might be less confusing for future contributors.

**Comment by ockham** [comment]
Rebased. Thanks for landing that whitespace PR, @simison!

File: packages/calypso-build/package.json (hunk 1 above)
Comment (sirreal) [reply packages/calypso-build/package.json]: I'm not sure so no need to change here 🙂

**Review by sirreal — COMMENTED** [review]
Working well 👍

**Pull request merged**